packages/x-components/src/x-modules/facets/components/facets/facets-provider.vue

```diff
@@ -19,97 +16,107 @@
-     * `Search X-Module` facets.
-     *
-     * @public
-     */
-    @Prop({ default: 'provided-facets' })
-    public groupId!: GroupId;
-
-    /**
-     * The facets to provide to the `Facets X-Module` state. They have to include the
-     * {@link @empathyco/x-types#Filter}.
-     *
-     * @internal
-     */
-    @Prop({ required: true })
-    public facets!: Facet[];
-
-    /**
-     * Temporarily stores the selected filters from the {@link FacetsProvider.facets} prop.
-     * This is necessary to handle the {@link FacetsXEvents.UserChangedSelectedFilters} event.
-     *
-     * @internal
-     */
-    protected selectedFilters: Filter[] | null = null;
-
-    /**
-     * A computed property to group the facets and the groupId. This is used by the watcher.
-     *
-     * @returns The FacetGroup with the facets and the group id.
-     *
-     * @internal
-     */
-    protected get facetsGroup(): FacetsGroup {
-      return { id: this.groupId, facets: this.facets };
-    }
-
-    /**
-     * Emits the {@link FacetsXEvents.UserChangedSelectedFilters} event when the user changes
-     * the selected filters.
-     *
-     * @param selectedFilters - The new list of selected filters.
-     * @internal
-     */
-    @XOn('SelectedFiltersChanged')
-    emitSelectedFiltersChanged(selectedFilters: Filter[]): void {
-      if (
-        this.selectedFilters === null ||
-        areFiltersDifferent(this.selectedFilters, selectedFilters)
-      ) {
-        this.$x.emit('UserChangedSelectedFilters', selectedFilters);
+  export default defineComponent({
+    name: 'FacetsProvider',
+    xModule: facetsXModule.name,
+    props: {
+      /**
+       * An facet group identifier to distinguish the provided facets from other facets like the
+       * `Search X-Module` facets.
+       *
+       * @public
+       */
+      groupId: {
+        type: String as PropType<GroupId>,
+        default: 'provided-facets'
+      },
+      /**
+       * The facets to provide to the `Facets X-Module` state. They have to include the
+       * {@link @empathyco/x-types#Filter}.
+       *
+       * @internal
+       */
+      facets: {
+        type: Array as PropType<Facet[]>,
+        required: true
       }
-      this.selectedFilters = null;
-    }
+    },
+    setup: function (props) {
+      const $x = use$x();
+
+      /**
+       * Temporarily stores the selected filters from the {@link FacetsProvider.facets} prop.
+       * This is necessary to handle the {@link FacetsXEvents.UserChangedSelectedFilters} event.
+       *
+       * @internal
+       */
+      const selectedFilters: Ref<Filter[] | null> = ref(null);
+
+      /**
+       * A computed property to group the facets and the groupId. This is used by the watcher.
+       *
+       * @returns The FacetGroup with the facets and the group id.
+       *
+       * @internal
+       */
+      const facetsGroup = computed(() => {
+        return { id: props.groupId, facets: props.facets };
```

Comment:
You can simplify this return statement with something like this: 
`const facetsGroup = computed(() => ({ id: props.groupId, facets: props.facets }));`